operator/src/main/java/oracle/kubernetes/operator/http/HttpAsyncRequestStep.java

```diff
@@ -108,7 +108,11 @@ private void checkTimeout(AsyncFiber fiber) {
 
     private void resume(AsyncFiber fiber, HttpResponse<String> response, Throwable throwable) {
       if (throwable != null) {
```

Comment:
This method now has two different tests for response being null - one in the block with throwable != null and one in the Optional.ofNullable(response) logic.

It would probably be better to merge them or otherwise simplify the logic. There are two different things being tested for null, here: `throwable` and `response`. Are all four combinations meaningful?

Reply:
> This method now has two different tests for response being null - one in the block with throwable != null and one in the Optional.ofNullable(response) logic.
> 
> It would probably be better to merge them or otherwise simplify the logic. There are two different things being tested for null, here: `throwable` and `response`. Are all four combinations meaningful?

The only permutation that does not meaningful is both null. Let me try to refactor this method a little.